src/streaming/TextSourceBuffer.js

```diff
@@ -91,37 +138,33 @@ MediaPlayer.dependencies.TextSourceBuffer = function () {
                 bytes = new Uint8Array(bytes);
                 ccContent=window.UTF8.decode(bytes);
                 try {
-                    result = self.getParser().parse(ccContent);
-                    label = mediaInfo.id;
-                    lang = mediaInfo.lang;
-                    self.getTextTrackExtensions().addTextTrack(self.videoModel.getElement(), result, label, lang, true);
-                    self.eventBus.dispatchEvent({type:MediaPlayer.events.TEXT_TRACK_ADDED});
+                    result = self.getParser(mimeType).parse(ccContent);
+                    createTextTrackFromMediaInfo(result, mediaInfo);
                 } catch(e) {
                     self.errHandler.closedCaptionsError(e, "parse", ccContent);
                 }
             }
         },
 
+        getIsDefault:function(mediaInfo){
+            return mediaInfo.index === this.mediaInfos[0].index; //TODO How to tag default. currently same order as in manifest. Is there a way to mark a text adaptation set as the default one?
+        },
+
         abort:function() {
-            this.getTextTrackExtensions().deleteCues(this.videoModel.getElement());
+            this.videoModel.getElement().textTracks.removeEventListener('change', onTextTrackChange);
+            this.textTrackExtensions.deleteAllTextTracks();
```

Comment:
It might be worth doing some checking of track.kind here, or at least ensuring that only tracks added by the TextSourceBuffer are deleted

In the future we may want to use track kind "metadata" to synchronise other types of data such as inband events with the video timeline and it would not be appropriate for the TextSourceBuffer to delete those. Tracks may also have been added automatically by the UA.